grabbit/src/main/groovy/com/twcable/grabbit/spring/batch/repository/JcrExecutionContextDao.groovy

```diff
@@ -306,4 +303,42 @@ class JcrExecutionContextDao extends AbstractJcrDao implements ExecutionContextD
         contextAsMap.each { key, value -> context.put(key, value) }
         context
     }
+
+    @Override
+    List<String> getJobExecutionContextPaths(List<String> jobExecutionResourcePaths) {
```

Comment:
@sagarsane lots of room for exceptions to be thrown in these methods. Should we be handling them in some way?

findResources throws:
```Throws:
QuerySyntaxException - If the query is not syntactically correct according to the query language indicator.
SlingException - If an error occurs querying for the resources.
IllegalStateException - if this resource resolver has already been closed.```

If any of the values that form the query are missing or can't be retrieved we'd have a problem and probably need to show what was missing or wrong with the query (if it can be determined). Same goes for the stepExecution method below.